agent/rpm/pbench-agent.spec.j2

```diff
@@ -8,42 +8,39 @@ URL:            https://github.com/distributed-system-analysis/pbench
 Source0:        pbench-agent-%{version}.tar.gz
 Buildarch:      noarch
 
-# EPEL provides ansible (a curated set of roles with a dependency on ansible-core)
-# on RHEL 8 and RHEL9. Only CentOS-Stream seems to not have an ansible package
-# available.
-
-%if 0%{?centos} >= 8
-Requires:  ansible-core
-%else
-Requires:  ansible
-%endif
-
 
 %if 0%{?rhel} == 7
-Requires:  python3, python3-pip
+Requires:  ansible
+%else
+Requires:  ansible-core
 %endif
 
-# This condition will be true on a CentOS-Stream-8 system as well
+# RHEL7, RHEL9 (and CentOS-Stream-9) and the Fedoras provide a python3 package
```

Comment:
We don't want to work with anything else by Python 3.8, with the exception of Python 3.whatever that is available on RHEL 7.

Reply:
On F35 and F36, doing `dnf module list` shows various versions of  perl, postgresql etc. but there is no mention of a different python.

Reply:
On my Fedora 36 box I see:

```
python3.11.x86_64                                         3.11.0~b1-2.fc36                      @updates
python3.11-devel.x86_64                                   3.11.0~b1-2.fc36                      @updates
python3.11-libs.x86_64                                    3.11.0~b1-2.fc36                      @updates
python3.11-tkinter.x86_64                                 3.11.0~b1-2.fc36                      @updates
python3.5.x86_64                                          3.5.10-7.fc34                         @updates
python3.6.x86_64                                          3.6.15-8.fc36                         @fedora
python3.7.x86_64                                          3.7.13-1.fc36                         @fedora
python3.8.x86_64                                          3.8.13-2.fc36                         @updates
python3.9.x86_64                                          3.9.13-1.fc36                         @updates
```

Reply:
Right, but were all of those there with F36 "out of the box", or did you install them at various times (perhaps pre-F36), e.g., when you were playing with Tox environments?

Reply:
Why does that matter for the Fedoras?  If they are available, but not installed, and a `pbench-agent` RPM has a `Requires: python3.8` on it, wouldn't they be installed along side the `pbench-agent` RPM?

Reply:
Yes..._if it is available_, then, yeah, I would expect `python3.8` to be installed as a precursor.

So, why does it seem to be available to you but not to Nick?

Reply:
What does your `/etc/yum.repos.d` look like? I have a fedora-modular.repo that is enabled, amd I just did a `dnf update` just in case, but I still can't see *any* python modules: `dnf module list` shows a bunch of other stuff, but nothing about python.  I cannot understand why the two setups are so different.

Reply:
That is because they are not modules.

```
$ sudo dnf list python3.\*.x86_64 | less -S
Last metadata expiration check: 1:22:15 ago on Sat 18 Jun 2022 06:06:11 AM EDT.
Installed Packages
python3.11.x86_64                       3.11.0~b1-2.fc36                @updates
python3.11-devel.x86_64                 3.11.0~b1-2.fc36                @updates
python3.11-libs.x86_64                  3.11.0~b1-2.fc36                @updates
python3.11-tkinter.x86_64               3.11.0~b1-2.fc36                @updates
python3.5.x86_64                        3.5.10-7.fc34                   @updates
python3.6.x86_64                        3.6.15-8.fc36                   @fedora
python3.7.x86_64                        3.7.13-1.fc36                   @fedora
python3.8.x86_64                        3.8.13-2.fc36                   @updates
python3.9.x86_64                        3.9.13-1.fc36                   @updates
Available Packages
python3.11.x86_64                       3.11.0~b3-1.fc36                updates
python3.11-debug.x86_64                 3.11.0~b3-1.fc36                updates
python3.11-devel.x86_64                 3.11.0~b3-1.fc36                updates
python3.11-idle.x86_64                  3.11.0~b3-1.fc36                updates
python3.11-libs.x86_64                  3.11.0~b3-1.fc36                updates
python3.11-test.x86_64                  3.11.0~b3-1.fc36                updates
python3.11-tkinter.x86_64               3.11.0~b3-1.fc36                updates
```

Reply:
Further, the existing `python3*` RPMs which provide our various python module dependency are only built for Python 3.10.  Notice that in the previous list `python3.10.*` does not exist.

If we install `python3.8`, and use that python in our invocations (that is another problem), then we'll still have to `pip3` install all our module dependencies.

We are going to have to support the distribution's default python version if we are going to avoid as many pip installs as possible.

Reply:
Resolving since we are now moving to Python 3.9 for RHEL 8 to line up with Python 3.9 being used in all other distributions except RHEL 7 (3.6) and Fedora 36 (3.10 default, but 3.9 available).